src/uint/boxed.rs

```diff
@@ -92,6 +94,17 @@ impl BoxedUint {
             .fold(Choice::from(1), |acc, limb| acc & limb.is_zero())
     }
 
+    /// Returns the truthy value if `self`!=0 or the falsy value otherwise.
+    pub(crate) fn is_nonzero(&self) -> Choice {
+        let mut b = 0;
+        let mut i = 0;
+        while i < self.limbs.len() {
+            b |= self.limbs[i].0;
+            i += 1;
+        }
+        Limb(b).is_nonzero().into()
```

Comment:
```suggestion
        let choice = Choice::from(1u8);
        
        for limb in &self.limbs {
            choice &= limb.ct_eq(&Limb::ZERO);
        }
        
        choice
```

Reply:
How about just `!self.is_zero()`?

Reply:
I mean, we can do that, but other instances can be replaced with `!x.is_zero()` too. It makes less sense when `const fn` isn't required